MPCAutofill/cardpicker/sources/api.py

```diff
@@ -19,15 +25,51 @@ class Folder:
     name: str
     parent: Optional["Folder"]
 
-    def get_full_path(self) -> str:
+    @functools.cached_property
+    def top_level_folder(self) -> "Folder":
         if self.parent is None:
-            return self.name
-        return f"{self.parent.get_full_path()} / {self.name}"
+            return self
+        return self.parent.top_level_folder
 
-    def get_top_level_folder(self) -> "Folder":
+    @functools.cached_property
+    def full_path(self) -> str:
+        _, name, _ = self.unpacked_name
         if self.parent is None:
-            return self
-        return self.parent.get_top_level_folder()
+            return name
+        return f"{self.parent.full_path} / {name}"
+
+    @functools.cached_property
+    def unpacked_name(self) -> tuple[pycountry.Languages, str, set[str]]:
+        """
+        The folder's name is unpacked according to the below schema. For example, consider `<EN> Cards [NSFW]`:
+             <EN>              Cards         [NSFW]
+        └─ language ──┘ └─ folder name ──┘ └─ tags ──┘
+        """
+
+        folder_name_results = re.compile(r"^(?:<(.+)> )?(.*?)$").search(self.name)
```

Comment:
Is there a reason to force the language at the start of the file name ? With the current regex, tags can be before or after the card name but not the language.

Reply:
i suppose you're right - i'll look into retooling this so the language specification can be anywhere in the name

Reply:
it seemed moderately annoying to extend this to retrieve the language from anywhere in the file name so i didn't end up doing it. enforcing that it must be at the start is straightforward enough for people to follow imo. might revisit this later.